presto-expressions/src/main/java/com/facebook/presto/expressions/translator/RowExpressionTranslator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.expressions.translator;
+
+import com.facebook.presto.spi.relation.CallExpression;
+import com.facebook.presto.spi.relation.ConstantExpression;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.SpecialFormExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+
+import static com.facebook.presto.expressions.translator.TranslatedExpression.untranslated;
+
+public class RowExpressionTranslator<T, C>
+{
+    public TranslatedExpression<T> translateConstant(ConstantExpression literal, C context, RowExpressionTreeTranslator<T, C> rowExpressionTreeTranslator)
+    {
+        return untranslated(literal);
+    }
+
+    public TranslatedExpression<T> translateVariable(VariableReferenceExpression reference, C context, RowExpressionTreeTranslator<T, C> rowExpressionTreeTranslator)
+    {
+        return untranslated(reference);
+    }
+
+    public TranslatedExpression<T> translateCall(CallExpression call, C context, RowExpressionTreeTranslator<T, C> rowExpressionTreeTranslator)
+    {
+        return translateExpression(call, context, rowExpressionTreeTranslator);
+    }
+
+    public TranslatedExpression<T> translateSpecialForm(SpecialFormExpression specialForm, C context, RowExpressionTreeTranslator<T, C> rowExpressionTreeTranslator)
+    {
+        return translateExpression(specialForm, context, rowExpressionTreeTranslator);
+    }
+
+    private TranslatedExpression<T> translateExpression(RowExpression expression, C context, RowExpressionTreeTranslator<T, C> rowExpressionTreeTranslator)
```

Comment:
Inline this function to `translateCall ` and `translateSpecialForm `

Reply:
Yeah, removing since translateExpression since we no longer do recursive translation.

As for merging RowExpressionTranslator and RowExpressionTree translator, I see the argument for why that may make sense since we do not do that much extra logic in tree translator anymore. However, in terms of an easy-to-understand interface for the developer I find that the way it is probably would work better since at a glance the developer knows what they need to extend/implement. Otherwise we would essentially be making them write their own `RowExpressionVisitor` if we merged these classes.

Reply:
Just remove it. Not that useful outside this class.